### DISCUSSION TITLE
Add flag for setting custom activation start time

### DISCUSSION THREAD

**Discussion by timhowellvista** [netr0m/az-pim-cli#78] [Ideas]
It would be great to have the ability to set custom activation date and start time (not just the duration)
e.g.
![image](https://github.com/user-attachments/assets/1c3f07d8-7718-4b35-8958-658dc12eac7e)

**Comment by netr0m** [comment]
Hi @timhowellvista ,

That should be doable. Are you able to share a snippet of the [request from the DevTools Network tab](https://learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/network/) with the date format included? (I'd guess it is ISO8601/RFC3339 formatted (e.g. `2024-12-02T10:08:06Z`)). Please make sure to sensor any sensitive information if you include other parts of the request 😄 Thanks!

**Comment by timhowellvista** [comment]
{"roleDefinitionId":xxxx","resourceId":xxxx","subjectId":"xxx","assignmentState":"Active","type":"UserAdd","reason":"Friday","ticketNumber":"1","ticketSystem":"SS"**,"schedule":{"type":"Once","startDateTime":"2024-12-05T19:00:00.000Z","endDateTime":null,"duration":"PT720M"}**,"linkedEligibleRoleAssignmentId":"xxxxx","scopedResourceId":""}

**Reply by netr0m** [reply]
Thanks! I've made a PR with a proposed solution, but due to lack of access to an Azure environment I am unable to test the functionality. Could you try out the development branch and see if it works for you? Thanks!

```sh
# Clone the branch for PR #81
git clone -b feat/set-start-time --single-branch https://github.com/netr0m/az-pim-cli.git az-pim-cli-tmp
cd az-pim-cli-tmp

# Build the binary
go build

# Run the binary
./az-pim-cli

# Run with --start-date for the date (DD/MM/YYYY) and --start-time for the time (HH:MM) (24 hour)
./az-pim-cli activate resource --name <RESOURCE_NAME> --role <RESOURCE_ROLE> --start-date 31/12/2024 --start-time 10:00 --debug

# Optionally clean up the tmp directory
rm -rf az-pim-cli-tmp
```

You can see the docs for the dev branch at https://github.com/netr0m/az-pim-cli/blob/feat/set-start-time/README.md#azure-resources-1

**Reply by timhowellvista** [reply]
Thanks. That looks like its working well:
![image](https://github.com/user-attachments/assets/90c7a0fd-c2d8-441e-8474-4d06af3bb3e3)
This was with --start-date "09/01/2025" --start-time "08:00"

**Comment by netr0m** [comment]
Tracked in #80

**Comment by netr0m** [comment]
Hi @timhowellvista ,

This has now been implemented and merged to the `main` branch. Will be included as part of the next release. Let me know if there's any issues. Thanks!

**Reply by netr0m** [reply]
Included in the latest release, `v1.6.0`: https://github.com/netr0m/az-pim-cli/releases/tag/v1.6.0
